core-rust/state-manager/src/transaction/validation.rs

```diff
@@ -74,7 +76,13 @@ impl LedgerTransactionValidator {
         let validator = NotarizedTransactionValidator::new(self.validation_config);
         match prepared_transaction {
             PreparedLedgerTransaction::User(notarized_transaction) => {
-                validator.validate(notarized_transaction, &self.intent_hash_manager)
+                // TODO: Remove
```

Comment:
Can you elaborate - how should it be removed in your opinion? Does this need changes in `radixdlt-scrypto`?
Or do you intend for the payload size to be calculated further up the stack and passed in here? In which case, can we do it as part of this PR?

Reply:
Yes, I think the fact that payload size is being passed around is a bit of a code smell suggesting that we have some weird abstractions. I'd prefer to keep this PR to just updating scrypto lib and working on another PR to fix the broken abstraction.